sdk/ai/azopenai/models_extra.go

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+package azopenai
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// ChatCompletionsOptionsFunctionCall - Controls how the model responds to function calls. "none" means the model does not
+// call a function, and responds to the end-user. "auto" means the model can pick between an end-user or calling a
+// function. Specifying a particular function via {"name": "my_function"} forces the model to call that function. "none" is
+// the default when no functions are present. "auto" is the default if functions
+// are present.
+type ChatCompletionsOptionsFunctionCall struct {
```

Comment:
Why were these moved from custom_models.go?

Reply:
(offline conversation - it's needed because of a chicken-and-egg situation with code gen)